### PR TITLE
Fix warning deprecate new URL in java 21

### PR DESCRIPTION
**Pull request by vttranlina**
- new URL -> new URI::toURL

ref: https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/net/URL.html#%3Cinit%3E(java.lang.String

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
```
07:28:11,144 [ERROR] /home/jenkins/jenkins-agent/workspace/james_ApacheJames_PR-2193/server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/json/SessionSerializationTest.scala:52: error: not found: type URI

07:28:11,144 [ERROR]   private val URL = new URI("http://james.org").toURL()
```
scala does not seem to like it somehow?

**Comment by vttranlina** [comment]
> 
> scala does not seem to like it somehow?

missing import 
my mistake when did by regex Intellij

File: server/container/core/src/main/java/org/apache/james/server/core/filesystem/ResourceFactory.java [hunk 1]
```diff
@@ -59,15 +57,15 @@ public Resource getResource(String fileURL) {
...
                 // No URL -> resolve as resource path.
                 return new ClassPathResource(fileURL);
             }
         }
     }
 
-    private Resource handleUrlResource(String fileURL) throws MalformedURLException, URISyntaxException {
-        URL url = new URI(fileURL).toURL();
+    private Resource handleUrlResource(String fileURL) throws MalformedURLException {
+        URL url = new URL(fileURL);
```
Comment (vttranlina) [inline server/container/core/src/main/java/org/apache/james/server/core/filesystem/ResourceFactory.java:68]: the logic for handler UrlResource is more complicated than I think -> I revert it now

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-2193/6/

looks related

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/SessionSupplier.scala [hunk 2]
```diff
@@ -80,11 +78,11 @@ class SessionSupplier(capabilityFactories: Set[CapabilityFactory]) {
 }
 
 class JmapUrlEndpointResolver(val urlPrefixes: UrlPrefixes) {
-  val apiUrl: URL = new URI(urlPrefixes.httpUrlPrefix.toString + "/jmap").toURL
+  val apiUrl: URL = URL(urlPrefixes.httpUrlPrefix.toString + "/jmap")
 
-  val downloadUrl: URL = new URI(urlPrefixes.httpUrlPrefix.toString + "/download/{accountId}/{blobId}?type={type}&name={name}").toURL
+  val downloadUrl: URL = URL(urlPrefixes.httpUrlPrefix.toString + "/download/{accountId}/{blobId}?type={type}&name={name}")
```
Comment (vttranlina) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/SessionSupplier.scala:83]: URI does not accept inputs containing {} characters. Upon examination, I verified that: apiUrl, downloadUrl, uploadUrl, eventSourceUrl, exclusively for session responses. Simple text (String) suffices for this purpose.

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**